### ISSUE TITLE
Question on Audit Log limitation in GCC

### ISSUE THREAD

**Issue by osbornecc** [microsoft/powerapps-tools#646]
Hello there,

With the _very_ recent ability to view Power Apps/Automate activities in the "new" compliance center (compliance.microsoft.com) in GCC, I was just wondering if there is a plan to get the audit log custom connector updated for GCC?

There is a documented limitation: https://docs.microsoft.com/en-us/power-platform/guidance/coe/limitations#government-community-cloud-gcc-environments

I'm just curious since it appears we should be able to get the info now.  (Full Disclosure:  I tried and while the flows run successfully and the custom connector tests fine, no data is pulled into the tables)
Thank you!

**Comment by junterpfe** [comment]
Happy to help test this in a GCC demo tenant if the DEVs need a place to try changes.

**Comment by manuelap-msft** [comment]
I don't think any update from our side is needed to get this working - if both the Audit Log and the API are available in GCC now, then this should work.

Can you help us do a quick test to see if the API is working as expected, before we move on to debugging why it's not fetching data? That would be very helpful.

1. Navigate to make.powerapps.com > Data > Custom Connector in the environment you have the CoE Audit solution installed
2. Edit the Office 365 Management API Audit Log connector that we shipped with the solution
3. Select 4. Test
4. Ensure a connection is selected - if you don't have a connection selected, click + New Connection to create one
5. Select the ListContent operation, enter your Tenant ID (you can find your Tenant ID in the Azure Portal) and a startTime and endTime. The Start Time and End Time have to be equal or less than 24 hours apart, so 2020-12-20 for startTime and 2020-12-21 for endTime should work
![image](https://user-images.githubusercontent.com/55705669/102887146-4c1c6800-444e-11eb-8588-eb3930d5cd5a.png)
6. Select Test Operation
7. Does this return a) Status 200 (which means success) and b) a long list of content data in the Body?
![image](https://user-images.githubusercontent.com/55705669/102887249-7d953380-444e-11eb-8f6c-46d2acf7eb6f.png)

Thank you for using the CoE, and offering to help us test this in a GCC tenant!

**Comment by osbornecc** [comment]
Hi there.  This is going to be long so I apologize up front :-)

**API Test Results**
First I show that the custom connector is set up to hit manage.office.com
![image](https://user-images.githubusercontent.com/50427725/102904168-32672a80-4426-11eb-907b-f99ea65ba1c4.png)

The Azure App Registration is configured with the documented API permissions.
![image](https://user-images.githubusercontent.com/50427725/102904489-a86b9180-4426-11eb-9705-b87e01f678c2.png)

Testing ListContent does return a 200 and a long list of values
![image](https://user-images.githubusercontent.com/50427725/102904686-ebc60000-4426-11eb-85ca-1324a1a48a89.png)

So, for this purpose, the test is successful.

**Finding PowerApps Activities (like LaunchPowerApp)**
Since running the delivered CoE audit log flows always return no data, I created a little test flow based off of what I saw in the delivered flows.  I wanted to see if I could return something people do every day:  LaunchPowerApp

Flow definition is shown below
![image](https://user-images.githubusercontent.com/50427725/102904891-38114000-4427-11eb-8b6d-9c4673707398.png)
![image](https://user-images.githubusercontent.com/50427725/102904926-495a4c80-4427-11eb-88cf-d5e5a5c63c54.png)

The flow runs successfully.  It finds 163 items in the log.  However, none of them have an operation of "LaunchPowerApp" when there should be hundreds.  They all seem to be tied to TeamsSessionStarted.  The below screenshot is one of the items.  It's not visible, but each entry includes multiple TeamsSessionStarted operations and sometimes a search alert one.
![image](https://user-images.githubusercontent.com/50427725/102905130-8b838e00-4427-11eb-83b9-95b8afca693b.png)


It seems that while I can hit the Audit Logs from the custom connector, the data isn't there.  But I can find it in compliance.microsoft.com.  This is what should be pulling into the connector for Dec 20-21.
![image](https://user-images.githubusercontent.com/50427725/102905717-68a5a980-4428-11eb-83b7-8eaff388b88f.png)


There is a disconnect somewhere, but I just don't know where.  I'm sure I'm missing something as simple as "checking a box"

Thank you for the help!

**Comment by manuelap-msft** [comment]
Thanks for the thorough testing! Now that we know that the Custom Connector returns Audit Logs, we can work on the Flow!

The Audit Log API has two limitations - one, it can't only return "Launched app" actions so we get all the Audit Log events and have to filter them down (as you've seen in the Filter action) and two, it can only return 100 Audit events at a time and doesn't paginate. That means we often have to retrieve the Audit Log events in smaller chunks to get all the data.

To retrieve the logs in smaller chunks, please open and edit the main Admin | Sync Audit Logs flow. Here, you will find a variable at the very top called Time Interval
![image](https://user-images.githubusercontent.com/55705669/102907417-ecce5000-446d-11eb-8b2b-23e423d240dc.png)

The Time Interval specifies the "chunks" in which we retrieve the Audit Log for the past day - if it's set to 1 Day, it will retrieve the Audit Log in one chunk for the past day. If your Audit Log contains more than 100 audit log events, it will only retrieve the Top 100. 

Try changing this Time Interval to 1 Hour (by changing Day to Hour) - this means that we will try and retrieve the Audit Log events in 24 chunks for the past day - so from 8:00 to 9:00, 9:00 to 10:00 etc. All the logic is taking care of in the flow - it calculates the time intervals and then calls the child flow the appropriate number of times for the chunks, so the only thing you will have to adjust is the Time Interval variable. 

Let's see if that helps when you try again! 

Sorry for the lengthy explanation - I wanted to make sure to explain both the logic in the flow, as well as the reason for asking you to make a change. Hope it all makes sense!

**Comment by osbornecc** [comment]
Ah, OK that is starting to make sense.  Thank you for the extra explanation!

After updating the time interval variable, running the Admin | Sync Audit Logs flow fails.  Before, it was running successfully but really quick (like 10 seconds).

The failure I'm running into is documented in #607, the hard-coded group ID.  There is a fix coming soon for that one!

At least, I think I'm now on the right track.  Should I wait for the next release, or is there a workaround for the group ID problem I could try?

Thanks again!

**Comment by osbornecc** [comment]
I apparently can't read today.  I saw the original poster's workaround in #607 and applied that same logic (replaced the group ID with one from my tenant).  The flow ran successfully with 1-hour chunks, but the Audit Log table still has no data.  I will play around with smaller chunks and see if I can get something.

Thank you

**Comment by osbornecc** [comment]
Oh my gosh.  I am quite the moron this week.

First, my account that owns CoE did not have the Security Reader Azure role.  (I have two accounts).  That was taken care of yesterday, but I still didn't see Power Platform activities.

Long story short, the custom connector has "mange.office.com" as the host.  That is not correct for GCC.  I modified the host per https://docs.microsoft.com/en-us/office/office-365-management-api/office-365-management-activity-api-reference?view=o365-worldwide#activity-api-operations.  Because the host changed, I had to restart the subscription in the Test section of the connector.

Now I have Power Platform activity data in the Audit Log table.  And at least I can say I was productive before the holiday :-)
